src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -93,46 +92,85 @@ internal ReflectTypeDescriptionProvider()
+            => LazyInitializer.EnsureInitialized(ref s_intrinsicTypeConverters, () => new Dictionary<object, IntrinsicTypeConverterData>(27)
         {
             // Add the intrinsics
             //
-            [typeof(bool)] = typeof(BooleanConverter),
-            [typeof(byte)] = typeof(ByteConverter),
-            [typeof(sbyte)] = typeof(SByteConverter),
-            [typeof(char)] = typeof(CharConverter),
-            [typeof(double)] = typeof(DoubleConverter),
-            [typeof(string)] = typeof(StringConverter),
-            [typeof(int)] = typeof(Int32Converter),
-            [typeof(short)] = typeof(Int16Converter),
-            [typeof(long)] = typeof(Int64Converter),
-            [typeof(float)] = typeof(SingleConverter),
-            [typeof(ushort)] = typeof(UInt16Converter),
-            [typeof(uint)] = typeof(UInt32Converter),
-            [typeof(ulong)] = typeof(UInt64Converter),
-            [typeof(object)] = typeof(TypeConverter),
-            [typeof(void)] = typeof(TypeConverter),
-            [typeof(CultureInfo)] = typeof(CultureInfoConverter),
-            [typeof(DateTime)] = typeof(DateTimeConverter),
-            [typeof(DateTimeOffset)] = typeof(DateTimeOffsetConverter),
-            [typeof(decimal)] = typeof(DecimalConverter),
-            [typeof(TimeSpan)] = typeof(TimeSpanConverter),
-            [typeof(Guid)] = typeof(GuidConverter),
-            [typeof(Uri)] = typeof(UriTypeConverter),
-            [typeof(Version)] = typeof(VersionConverter),
+            [typeof(bool)] = new IntrinsicTypeConverterData((type) => new BooleanConverter()),
+            [typeof(byte)] = new IntrinsicTypeConverterData((type) => new ByteConverter()),
+            [typeof(sbyte)] = new IntrinsicTypeConverterData((type) => new SByteConverter()),
+            [typeof(char)] = new IntrinsicTypeConverterData((type) => new CharConverter()),
+            [typeof(double)] = new IntrinsicTypeConverterData((type) => new DoubleConverter()),
+            [typeof(string)] = new IntrinsicTypeConverterData((type) => new StringConverter()),
+            [typeof(int)] = new IntrinsicTypeConverterData((type) => new Int32Converter()),
+            [typeof(short)] = new IntrinsicTypeConverterData((type) => new Int16Converter()),
+            [typeof(long)] = new IntrinsicTypeConverterData((type) => new Int64Converter()),
+            [typeof(float)] = new IntrinsicTypeConverterData((type) => new SingleConverter()),
+            [typeof(ushort)] = new IntrinsicTypeConverterData((type) => new UInt16Converter()),
+            [typeof(uint)] = new IntrinsicTypeConverterData((type) => new UInt32Converter()),
+            [typeof(ulong)] = new IntrinsicTypeConverterData((type) => new UInt64Converter()),
+            [typeof(object)] = new IntrinsicTypeConverterData((type) => new TypeConverter()),
+            [typeof(CultureInfo)] = new IntrinsicTypeConverterData((type) => new CultureInfoConverter()),
```

Comment:
The only types in here that can be inherited from are `CultureInfo` and `Uri`. If someone created a class that inherited from `CultureInfo` and then tried looking up a converter for it, previously this would be found. But with the new changes they won't be.  (The previous implementation would search up the type hierarchy until it found an entry.)

We should probably add tests for this as well.